packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for invoking an API Endpoint with ApiGatewayInvoke
+ */
+export interface ApiGatewayInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * hostname of an API Gateway URL
+   * @example {ApiId}.execute-api.{region}.amazonaws.com
+   */
+  readonly apiEndpoint: string;
+
+  /**
+   * Http method for the API
+   */
+  readonly method: HttpMethod;
+
+  /**
+   * HTTP headers string to list of strings
+   * @default - No headers
+   */
+  readonly headers?: sfn.TaskInput;
```

Comment:
does this need to be a `TaskInput`? - is that because they can be passed through the state input as too?

Reply:
It's just a json. I can change the type.